src/ui/NoNameFacet/NoNameFacetValues/NoNameFacetValueElement.ts

```diff
@@ -0,0 +1,109 @@
+import { $$ } from '../../../utils/Dom';
+import { NoNameFacet } from '../NoNameFacet';
+import { NoNameFacetValueCheckbox } from './NoNameFacetValueCheckbox';
+import { NoNameFacetValue } from './NoNameFacetValue';
+import { NoNameFacetValueUtils } from './NoNameFacetValueUtils';
+import { l } from '../../../strings/Strings';
+import { KeyboardUtils, KEYBOARD } from '../../../utils/KeyboardUtils';
+
+export class NoNameFacetValueElement {
+  public element: HTMLElement;
+  private labelElement: HTMLElement;
+  private labelWrapperElement: HTMLElement;
+  private captionElement: HTMLElement;
+  private countElement: HTMLElement;
+  private checkboxElement: HTMLElement;
+
+  constructor(private facet: NoNameFacet, private facetValue: NoNameFacetValue) {
+    this.create();
+  }
+
+  private create() {
+    this.createContainer();
+    this.createAndAppendLabel();
+    this.createAndAppendLabelWrapper();
+    this.createAndAppendCheckbox();
+    this.createAndAppendCount();
+    this.createAndAppendCaption();
+
+    this.toggleSelectedClass();
+    this.addFocusAndBlurEventListeners();
+    this.addClickEventListeners();
+  }
+
+  private createContainer() {
+    this.element = $$('li', {
+      className: 'coveo-facet-value coveo-facet-selectable',
+      dataValue: this.facetValue.value
+    }).el;
+  }
+
+  private createAndAppendLabel() {
+    this.labelElement = $$('label', { className: 'coveo-facet-value-label' }).el;
+    this.element.appendChild(this.labelElement);
+  }
+
+  private createAndAppendLabelWrapper() {
+    this.labelWrapperElement = $$('div', { className: 'coveo-facet-value-label-wrapper' }).el;
+    this.labelElement.appendChild(this.labelWrapperElement);
+  }
+
+  private createAndAppendCheckbox() {
+    this.checkboxElement = new NoNameFacetValueCheckbox(this.label).element;
+    this.labelWrapperElement.appendChild(this.checkboxElement);
+  }
+
+  private createAndAppendCount() {
+    this.countElement = $$('span', { className: 'coveo-facet-value-count' }, this.count).el;
+    this.labelWrapperElement.appendChild(this.countElement);
+  }
+
+  private createAndAppendCaption() {
+    this.captionElement = $$(
+      'span',
+      {
+        className: 'coveo-facet-value-caption',
+        title: this.caption,
+        dataOriginalValue: this.facetValue.value
+      },
+      this.caption
+    ).el;
+    this.labelWrapperElement.appendChild(this.captionElement);
+  }
+
+  private toggleSelectedClass() {
+    $$(this.element).toggleClass('coveo-selected', this.facetValue.selected);
+  }
+
+  private addFocusAndBlurEventListeners() {
+    $$(this.checkboxElement).on('focus', () => $$(this.element).addClass('coveo-focused'));
+    $$(this.checkboxElement).on('blur', () => $$(this.element).removeClass('coveo-focused'));
+  }
+
+  private addClickEventListeners() {
+    $$(this.element).on('click', this.selectAction);
```

Comment:
I think there would be a way to simply have a :
```
<label><input type="chechbox" />The facet value (count)</label>
```
without having to manually add all those events in javascript ?
Need to make sure we are using the correct semantic HTML, and not div/span everywhere.

Reply:
Before putting the count inside parentheses and stuff I would wait for the more final mockups for the new facets.